benchmarks/media-streaming/client/files/videoperf/core.c

```diff
@@ -312,7 +312,10 @@ port_get (void)
 		mask = 0;
 	else
 		mask = ~((1UL << (bit + 1)) - 1);
-	port_free_map[i] &= ~(1UL << bit);
+  // commented by aansaarii
```

Comment:
I can't say I understand this change, it's not obvious to me what is the problem in the code. Why does your change to _not_ set bits in the port map solve it? It seems like all that this change does is "shoot the messenger" for lack of a better term, because the function `get_port()` can still return a valid answer without setting the port map structure.

Reply:
Did not the comment that I wrote above clarify the need for this change? If httperf sets the bit here and later on, it notices that it cannot bind the connection to the found port, it will never call `core_close` that calls `put_port` that resets the port bit to make it available for the future connections. The `port_get` function has to only suggest a port from its free list and setting the bit has to be done after httperf detects no other process on the machine uses that port to safely allocate it (which I did it [here](https://github.com/parsa-epfl/cloudsuite/blob/fix_streaming_server/benchmarks/media-streaming/client/files/videoperf/core.c#L1179)) and then, release the port by resetting the bit whenever the connection is closed.

Reply:
After discussion, this change only marks ports as allocated when _this process_ successfully binds to them.

The implications are that other processes may have bound to ports that this process believes are available, and therefore more syscalls to open ports may fail. This may or may not impact performance, which needs to be tested.